### PR TITLE
feat: execute command parsing and gen

### LINKED ISSUE #13: `execute` command handling
create handling for execute command

### PR DESCRIPTION
**Pull request by feniljain**
Hey @tjdevries :wave:,

A small try to fix #13

### Doubt

How to generate the outputs from the snapshots? ( hence also not been able to test this code )

### Truth
 
 I am not even remotely sure if this is the correct way to do it, so pardon me if it's all wrong 🙈

### CODE REVIEW

**Comment by tjdevries** [comment]
To generate new snapshots, just run `cargo test` in the crate that you're working on. If they update, then you need to do `cargo insta review` (probably need to first do `cargo install insta`) and then go through the prompts.

File: crates/vim9-gen/src/lib.rs [hunk 1]
```diff
@@ -741,6 +741,12 @@ impl Generate for EchoCommand {
     }
 }
 
+impl Generate for ExecuteCommand {
+    fn gen(&self, state: &mut State) -> String {
+        format!("vim.fn.execute({})", self.expr.gen(state))
```
Comment (tjdevries) [inline crates/vim9-gen/src/lib.rs]: I think there is an API command (maybe vim.api.nvim_exec, don't remember offhand) that we might want to use instead (in general, I prefer using API)

**Comment by tjdevries** [comment]
The only other thing to test would be checking if you can handle each of the situations described by `:help :execute`

**Comment by tjdevries** [comment]
Also, add a new test for execute by adding a new file and then adding a test w/ the other ones that loads that file

File: crates/vim9-gen/src/lib.rs (hunk 1 above)
Comment (feniljain) [reply crates/vim9-gen/src/lib.rs]: `vim.api.nvim_command` seems to be the perfect option, so shifting to that :)

**Comment by feniljain** [comment]
Heyo,

One small doubt, are we supporting syntax like this?

```vim
execute '!ls' | echo "theend"
```

If yes then we need to fix it cause currently I get `TokenKind` as `Illegal` for `Or` here, we would need to add support for the same, and I guess that would be better done in a separate PR and for this one, I can support cases:

```vim
execute "normal ixxx\<Esc>"
execute "!ls " .. shellescape('./rustfmt.toml', 1)
```

**Comment by tjdevries** [comment]
Yeah, we can definitely do more improvements later, I wouldn't worry about the `|` case yet -- I will have to think about how to support that.

**Comment by tjdevries** [comment]
(and it looks like I created a conflict in some of the stuff I pushed up so far, so feel free to push and fix the conflict and then I'll review again)

**Comment by tjdevries** [comment]
sweet! LGTM :)

I'll merge. If you want to do something else, let me know. I'm changing a bunch of stuff locally right now but will push up in the next day or so

**Comment by feniljain** [comment]
Yeah sure, I will go through the list of issues, and try to pick something up :) ( after your push ofc )

**Comment by tjdevries** [comment]
The next best one to do would be for remap style items. Working on parsing them would be awesome. You could start w/ just parsing nnoremap (and I could work out how we want to match shortened versions and all that good stuff later)

**Pull request merged**